tests/PullRequest.py

```diff
@@ -402,3 +407,70 @@ def testUpdateBranch(self):
             self.pull.update_branch("addaebea821105cf6600441f05ff2b413ab21a36")
         )
         self.assertTrue(self.pull.update_branch())
+
+    def testDeleteOnMerge(self):
+        self.assertTrue(
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        )
+        self.assertFalse(self.delete_restore_pull.is_merged())
+        status = self.delete_restore_pull.merge(deletebranch=True)
+        self.assertTrue(status.merged)
+        self.assertTrue(self.delete_restore_pull.is_merged())
+        with self.assertRaises(github.GithubException) as raisedexp:
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        self.assertEqual(raisedexp.exception.status, 404)
+        self.assertEqual(
+            raisedexp.exception.data,
+            {
+                "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-branch",
+                "message": "Branch not found",
+            },
+        )
+
+    def testRestoreBranch(self):
+        with self.assertRaises(github.GithubException) as raisedexp:
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        self.assertEqual(raisedexp.exception.status, 404)
+        self.assertEqual(
+            raisedexp.exception.data,
+            {
+                "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-branch",
+                "message": "Branch not found",
+            },
+        )
+        self.assertTrue(self.delete_restore_pull.restore_branch())
+        self.assertTrue(
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        )
+
+    def testDeleteBranch(self):
+        self.assertTrue(
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        )
+        self.assertIs(self.delete_restore_pull.delete_branch(force=False), None)
+        with self.assertRaises(github.GithubException) as raisedexp:
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        self.assertEqual(raisedexp.exception.status, 404)
+        self.assertEqual(
+            raisedexp.exception.data,
+            {
+                "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-branch",
+                "message": "Branch not found",
+            },
+        )
+
+    def testForceDeleteBranch(self):
```

Comment:
Are any PRs linked to a force deleted branch also deleted? I'd be interested to see if that is the case.

Reply:
Yes, a PR with the head ref being the branch that is deleted gets closed.

Reply:
Oops.. I tested with the branch of this PR 🤦